ca_description/urdf/sensors/color_sensor.xacro

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
+  <xacro:include filename="$(find ca_description)/urdf/common_properties.xacro"/>
+
+  <xacro:macro name="color_sensor" params="name_prefix color parent visualize:=false *origin">  
+    <xacro:property name="parent_link" value="${parent}"/>
+    
+    <link name="${name_prefix}_color_sensor">
+        <visual>
+            <geometry>
+                <mesh filename="package://ca_description/meshes/scaled_lens.stl"/>
+            </geometry>
+        </visual>
+    </link>
+
+    <joint name="base_to_${name_prefix}_color_sensor" type="fixed">
+        <parent link="${parent_link}"/>
+        <child link="${name_prefix}_color_sensor"/>
+        <xacro:insert_block name="origin" />
+    </joint>
+
+    <joint name="${name_prefix}_color_optical_joint" type="fixed">
+        <origin xyz="0.05 0 0" rpy="${-pi/2} 0 ${-pi/2}" />
+      <parent link="${name_prefix}_color_sensor" />
+      <child link="${name_prefix}_color_optical_frame"/>
+    </joint>
+    
+    <link name="${name_prefix}_color_optical_frame"/>
+    
+    <gazebo reference="${name_prefix}_color_sensor">
+        <sensor name='camera' type='camera'>
+        <camera name='__default__'>
+         <horizontal_fov>0.3</horizontal_fov>
+         <image>
+           <width>120</width>
```

Comment:
I guess this can be even smaller, ideally, you will need a 1x1 image frame.

Reply:
Adressed on the PR referenced below